### ISSUE TITLE
1.20.1 Forge Redstone differs from Vanilla and Fabric

### ISSUE THREAD

**Issue by eeble1** [MinecraftForge/MinecraftForge#9973]
**Minecraft Version:** {1.20.1}

**Forge Version:** {Forge version. *47.2*}

**Steps to Reproduce:**
 1. download the linked world (Orbital Strike 4.0)
 2. use the cannon
 3. set the command blocks to use @ A instead of the username present
 4. go to spectator and watch the bottom TNT duper (first one to activate)

**Description of issue:**
It seems that both the timing as well as the normal game features differ in forge as whenever I use the cannon the timing for pushing the TNT is off as well as whatever TNT does get put on the cannon never makes it to target.

**Link to Files**
https://www.mediafire.com/folder/88qrgise22lax/Orbital_Strike_Cannon

**Comment by LexManos** [comment]
To debug this I need something far simpler then this massive contraption. At the end of the day its the TNT duplicator that is the issue, so find a isolated case to test that.

**Comment by eeble1** [comment]
I will try to create a simpler version of the mechanism used in the cannon later. I also want to ask whether the main mechanisms behind the cannon such as building momentum on the TNT stuck in stasis would still work on forge and whether it would make its way to target or whether the main mechanisms behind it are broken by forge. if the mechanisms work on forge what I could do is fine tune the cannon in order to make it work within forge.

**Comment by LexManos** [comment]
Chunk loading is modified by Forge, however in theory it should use the same chunk ticketing system as vanilla. Just with the ability for mods to add extra ticket types. 
So.. in theory it should be fine. 
The issue is that this entire process exploits bugs in Minecraft. Sometimes bugs are fixed by Forge. But typically with intentional fixes and links to the bug reports on Mojang's tracker. I can't say if any of those changes interact with anything in this specific contraption. Which is why smaller reproduction cases are needed.

Also for future note, a bit more detailed reproduction steps is also helpful. 
We had no idea that `use the cannon` meant that we needed to grab at least a trim x, trim z, and payload type from the chest, add it to the minecart right click one note block, then right click another. I only found that out by googeling the name that was in the command blocks and seeing what I could find.

`set the command blocks to use @ A instead of the username present` also was slightly better because we only found 3 command blocks, but what we actually had to do was edit the two touching each other and then add a button to the pair. and then press that button after starting the firing sequence {the 2nd note block}

And then `go to spectator and watch the bottom TNT duper (first one to activate)`
Took us feakin ages to figure out where the tnt duper even was, as the contraption is quite large, and it does literally nothing unless you do the first step correctly {which again we had no way of knowing how to do}
I was able to find a TNT duper explanation [here ](https://youtu.be/65OwMFbnHnE) and [here](https://youtu.be/mM0uO_BGyEo?t=240).  But none that looked anything like what your setup is.

Overall, my main point is that when reporting issues, its nice to have a example of the issue but it needs to be something A LOT simpler and better explained.  So even tho we tried, and it was kinda fun digging into it, we had no idea how to isolate the issue and verify it. Which is needed to be done before we can fix it.

**Comment by Jonathing** [comment]
This issue is becoming stale. If you're still interested, please send us a reproducible test case so we can try and fix whatever's causing this.

**Comment by VoidLeech** [comment]
Here's two detailed Neoforge issues on comparator behaviour differing that might be relevant as the analyzed code (seems to) originates(s) from before they forked from Forge:
https://github.com/neoforged/NeoForge/issues/1750 (missing updates)
https://github.com/neoforged/NeoForge/issues/1863 (different update order)

**Comment by Jonathing** [comment]
The write-ups are useful, but [like mentioned earlier](https://github.com/MinecraftForge/MinecraftForge/issues/9973#issuecomment-2113108713), having an isolated, reproducible test case is going to help us actually understand what the issue is with Forge and how we can actively tinker with it without causing much trouble.

**Comment by LexManos** [comment]
Fixed issues related to the update ordering and the up/down comparator issues. The 'missing updates' related to the simulator aspect of item transferring I'll look into later. Would be good to have a simplified test case/structure for it.

**Comment by Jonathing** [comment]
I mentioned this in #10417, but I wanted to also mention here that while we will be working on cleaning up the differences in Redstone between Forge and Vanilla, I don't think I am comfortable with backporting the changes we make to 1.20.6 and older. As I mentioned, since Forge for 1.20.6 and older is stable, I fear (reasonably) that making those changes would cause all kinds of grief.

If you think that there is good enough reason to consider otherwise, let me know and I'll hear you out. But as it stands, whatever changes we make to bridge parity for redstone will not be backported further than 1.21.1. The reason I mention this at all is because this request was made for 1.20.1.

**Comment by LexManos** [comment]
Honestly this is a double edged sword, as this seems to be a bug since the introduction of this feature which is like 10 years ago.
So there could be some contraptions that rely on the 'bugged' redstone order. And some that rely on vanilla's order.
This is the same anger that Moajng gets any time they touch redstone stuff. People thinking that bugs are features.
The PROPER way to do it would probably be to add it as a config option. But I don't want to do that.
I'm fine with backporting to match vanilla tho, so feel free to backport.

**Comment by Jonathing** [comment]
If I were to backport this I *would* add a config option that is disabled by default. So that's probably what I would do. After all, server configs (are supposed to!!!) sync to clients.
